### PR TITLE
[ENG-5140] #2 Update get_auth for GV and readability 

### PR DESCRIPTION
**Pull request by Johnetordoff**
## Purpose

Fix some signals code that wasn't covered by tests.

## Changes

- squish some ORM logic to avoid errors

## QA Notes

Please make verification statements inspired by your code and what your code touches.
- Verify
- Verify

What are the areas of risk?

Any concerns/considerations/questions that development raised?

### CODE REVIEW

File: addons/base/views.py [hunk 1]
```diff
@@ -697,22 +697,23 @@ def osfstoragefile_update_view_analytics(self, auth, fileversion):
 
 @file_signals.file_viewed.connect
 def osfstoragefile_viewed_update_metrics(self, auth, fileversion):
-    resource = BaseFileNode.objects.get(versions__id=fileversion.id).target
+    file = BaseFileNode.objects.get(versions__id=fileversion.id)
```
Comment (jwalz) [inline addons/base/views.py:700]: still a problem with `get` possibly returning multiple results

File: addons/base/views.py [hunk 2]
```diff
@@ -697,22 +697,23 @@ def osfstoragefile_update_view_analytics(self, auth, fileversion):
 
 @file_signals.file_viewed.connect
 def osfstoragefile_viewed_update_metrics(self, auth, fileversion):
-    resource = BaseFileNode.objects.get(versions__id=fileversion.id).target
+    file = BaseFileNode.objects.filter(versions__id=fileversion.id).last()
```
Comment (mfraezz) [inline addons/base/views.py:700]: `.last` is guaranteed to not throw a `MultipleObjectsReturned` error, but isn't guaranteed to deliver the correct `FileNode` object. Would it be possible to include it in the `file_signals.file_viewed.send` call instead?

File: addons/base/views.py [hunk 3]
```diff
@@ -697,22 +697,23 @@ def osfstoragefile_update_view_analytics(self, auth, fileversion):
...
+    resource = file.target
     if waffle.switch_is_active(features.ELASTICSEARCH_METRICS) and isinstance(resource, Preprint):
         try:
             PreprintView.record_for_preprint(
                 preprint=resource,
                 user=auth.user,
                 version=fileversion.identifier,
-                path=fileversion.path,
```
Comment (mfraezz) [inline addons/base/views.py:707]: There's at least one other instance of this that needs fixing in this file.

File: addons/base/views.py (hunk 2 above)
Comment (Johnetordoff) [reply addons/base/views.py:700]: Won't all the versions have the same BaseFIleNode though? I only retrieve the version, not the full node.

Comment (jwalz) [reply addons/base/views.py:700]: FileVersions help dedupe information, so one FileVersion can belong to many BaseFileNodes

File: addons/base/views.py [hunk 4]
```diff
@@ -677,58 +688,56 @@ def addon_delete_file_node(self, target, user, event_type, payload):
...
         # mark fileversion as seen
         FileVersionUserMetadata.objects.get_or_create(user=auth.user, file_version=fileversion)
 
 
 @file_signals.file_viewed.connect
-def osfstoragefile_update_view_analytics(self, auth, fileversion):
-    file = BaseFileNode.objects.get(versions__id=fileversion.id)
-    node = file.target
+def osfstoragefile_update_view_analytics(self, auth, fileversion, file_node):
+    node = file_node.target
```
Comment (jwalz) [inline addons/base/views.py:699]:
```suggestion
    resource = file_node.target
```

File: addons/base/views.py [hunk 5]
```diff
@@ -677,58 +688,56 @@ def addon_delete_file_node(self, target, user, event_type, payload):
...
     enqueue_update_analytics(
         node,
```
Comment (jwalz) [inline addons/base/views.py:701]:
```suggestion
        resource,
```

**Review by jwalz — APPROVED** [review]
One tiny nit

**Pull request merged**